runtime/src/asset.rs

```diff
@@ -244,78 +237,13 @@ decl_module! {
-            let sender = ensure_signed(origin)?;
-            let sender_key = Key::try_from(sender.encode())?;
-            let from_did =  match <identity::Module<T>>::current_did() {
-                Some(x) => x,
-                None => {
-                    if let Some(did) = <identity::Module<T>>::get_identity(&sender_key) {
-                        did
-                    } else {
-                        return Err("did not found");
-                    }
-                }
-            };
-
-            let ticker = utils::bytes_to_upper(_ticker.as_slice());
-
-            ensure!(!<Tokens<T>>::exists(&ticker), "token already created");
-
-            ensure!(Self::is_ticker_registry_valid(&ticker, from_did), "ticker registered to someone else");
-
-            let to_did_ticker = (to_did, Some(ticker.clone()));
-
-            if <TickerTransferApprovals>::exists(&to_did_ticker) {
-                <TickerTransferApprovals>::mutate(&to_did_ticker, |tta| tta.authorized_by = from_did);
-            } else {
-                let to_did_none = (to_did, None);
-                let next_ticker;
-                if <TickerTransferApprovals>::exists(&to_did_none) {
-                    let none_tta = Self::ticker_transfer_approvals(&to_did_none);
-                    next_ticker = none_tta.next_ticker.clone();
-                    if next_ticker.is_some() {
-                        <TickerTransferApprovals>::mutate(
-                            (to_did, none_tta.next_ticker),
-                            |tta| tta.previous_ticker = Some(ticker.clone())
-                        );
-                    }
-                } else {
-                    next_ticker = None;
-                }
-
-                let none_tta = TickerTransferApproval {
-                    authorized_by: from_did,
-                    next_ticker: Some(ticker.clone()),
-                    previous_ticker: None,
-                };
-                <TickerTransferApprovals>::insert(&to_did_none, none_tta);
-
-                let tta = TickerTransferApproval {
-                    authorized_by: from_did,
-                    next_ticker: next_ticker,
-                    previous_ticker: None,
-                };
-                <TickerTransferApprovals>::insert(&to_did_ticker, tta);
-            }
-
-            Self::deposit_event(RawEvent::TickerTransferApproval(ticker, from_did, to_did));
-
-            Ok(())
-        }
-
-        /// This function is used to transfer a ticker to someone else
-        ///
-        /// # Arguments
-        /// * `origin` It consist the signing key of the caller (i.e who signed the transaction to execute this function)
-        /// * `_ticker` ticker to transfer
-        pub fn process_ticker_transfer(origin, _ticker: Vec<u8>) -> Result {
+        /// * `auth_id` Authorization ID of ticker transfer authorization
+        pub fn accept_ticker_transfer(origin, auth_id: u64) -> Result {
```

Comment:
Would it make sense to have a function (maybe in identity.rs) which is:
`accept_authorisation(origin, auth_id)`
which then looks at the associated AuthorisationData enum and routes to the appropriate underlying function with the right business logic to execute it?

Reply:
This could then be batched, where we take a Vec<auth_id> and execute them as a batch.